src/analyzer/analyze.server.ts

```diff
@@ -148,6 +149,7 @@ async function analyzeTree(path: string, name: string, hash: string) {
           name: newName,
           sizeInBytes: child.size as number,
           blameAuthors: {},
+          mutex: new Mutex()
```

Comment:
Is the mutex necessary when there is no multithreading?

Reply:
Not sure, i guess it doesn't hurt. Haven't bothered to test O_O

Reply:
Just tested, it does not matter :(
it just seems way cooler with a mutex there